### PR TITLE
add atom to possible editors

### PR DESCRIPTION
**Pull request by nplusp**

### CODE REVIEW

**Comment by WizardOfOgz** [comment]
@nplusp What are using to handle URLs with the atom:// protocol and have files opened in Atom?

**Comment by RobinDaugherty** [comment]
This is an attempt at #314. Unfortunately [Atom doesn't handle `atom://` URLs](https://github.com/atom/atom/issues/2037) so until that's fixed, this change seems to be the wrong idea. We could instead implement the `atm://` scheme, but that's only supported with a plugin, and then what will we do once Atom fixes their issue?

**Comment by RobinDaugherty** [comment]
Instructions for the interim solution are [in the wiki](https://github.com/charliesome/better_errors/wiki/Link-to-your-editor#atom) for now.

**Comment by shanept** [comment]
Hi guys,

I'm currently working on this feature for Atom. See atom/atom#14243

Thanks

**Comment by UziTech** [comment]
This pull request should work in Atom v1.23.0 (which is in beta right now)

https://github.com/charliesome/better_errors/wiki/Link-to-your-editor#atom

File: lib/better_errors.rb [hunk 1]
```diff
@@ -13,12 +13,13 @@
 
 module BetterErrors
   POSSIBLE_EDITOR_PRESETS = [
+    { symbols: [:atom],                 sniff: /atom/i,  url: "atom://open?url=file://%{file}&line=%{line}" }
```
Comment (UziTech) [inline lib/better_errors.rb]: needs a comma at the end

**Comment by nplusp** [comment]
@UziTech rebased on master and added comma

**Comment by BinaryMuse** [comment]
Starting in Atom 1.23, this will be supported in Atom core via https://github.com/atom/atom/pull/15935. The URL format is `atom://core/open/file?filename=urlEncodedFileName&line=n&column=n`. The primary benefit of using the core handler instead of a third-party package is that the core handler will attempt to search all windows to try to find the one that most likely contains the file in question.

**Comment by nplusp** [comment]
@BinaryMuse thanks! I've updated my PR respectively. ❤️🎉

**Comment by RobinDaugherty** [comment]
Since this version of Atom is still in beta (and there's a minor inconsistency with line number handling) I've updated [the wiki page](https://github.com/charliesome/better_errors/wiki/Link-to-your-editor#atom) to include instructions for the current beta.

**Comment by BinaryMuse** [comment]
@RobinDaugherty I've merged a fix to the off-by-one issue into Atom master and into the current beta branch, and it will ship with the next beta hotfix (or the next stable release, whichever comes first).

One quick note: URI handling for the `atom://` protocol doesn't work prior to Atom 1.23, even with third-party packages installed (as opposed to what the current wiki article states).

**Comment by RobinDaugherty** [comment]
Thanks @BinaryMuse! I've updated the wiki page to reflect that.

**Pull request merged**